sktalk/corpus/parsing/parser.py

```diff
@@ -1,9 +1,24 @@
 import abc
+import datetime
+from ..conversation import Conversation
 
 
 class Parser(abc.ABC):
     """Abstract parser class."""
 
     @abc.abstractmethod
-    def parse(self, file) -> "Conversation":  # noqa: F821
+    def parse(self, file) -> Conversation:
         return NotImplemented
+
+    @staticmethod
+    def _to_timestamp(time_ms):
+        try:
+            time_ms = float(time_ms)
+        except ValueError:
+            return None
```

Comment:
Is this expected to happen in real data? Otherwise, the exception should be raised.

Reply:
Not sure what generated this particular error but I'm afraid it does sometimes happen that we don't have all timestamps (e.g. some source data sets come with some annotations that have `begin` but not `end`, in which case we also cannot compute `duration`). However, this is problematic for downstream analyses so I think we would want to raise an error or at least a warning

Reply:
Ah, makes sense. I would then raise the error here and handle it downstream:
```suggestion
     except ValueError as e:
            raise ValueError("Could not parse timestamp: '{str(time_ms)}'") from e
```

Reply:
Seems reasonable — can you weigh in @liesenf?